tests/v2/validation/provisioning/rke2/agent_customization_test.go

```diff
@@ -66,26 +65,10 @@ func (r *RKE2AgentCustomizationTestSuite) SetupSuite() {
 }
 
 func (r *RKE2AgentCustomizationTestSuite) TestProvisioningRKE2ClusterAgentCustomization() {
-	productionPool := []provisioninginput.MachinePools{
-		{
-			NodeRoles: machinepools.NodeRoles{
-				ControlPlane: true,
-				Quantity:     2,
-			},
-		},
-		{
-			NodeRoles: machinepools.NodeRoles{
-				Etcd:     true,
-				Quantity: 3,
-			},
-		},
-		{
-			NodeRoles: machinepools.NodeRoles{
-				Worker:   true,
-				Quantity: 2,
-			},
-		},
-	}
+	productionPool := []provisioninginput.MachinePools{provisioninginput.EtcdMachinePool, provisioninginput.ControlPlaneMachinePool, provisioninginput.WorkerMachinePool}
+	productionPool[0].MachinePoolConfig.Quantity = 3
+	productionPool[1].MachinePoolConfig.Quantity = 2
+	productionPool[2].MachinePoolConfig.Quantity = 2
```

Comment:
Are we leaving worker nodes at 2, or do we want this at 3?

Reply:
imo 2 is okay. 3 is more standard but I wanted to save on resources since this will be a more consistently ran test